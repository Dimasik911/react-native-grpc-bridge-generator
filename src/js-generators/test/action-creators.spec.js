/* eslint-env jest */

import generator, { generateFileHeader, generateActionCreator } from '../action-creators'

describe('generateActionCreator', () => {
  it('creates a repeated type definition', () => {
    const rsp = generateActionCreator('GetUser', 'GetUser')

    expect(rsp).toMatchSnapshot()
    expect(rsp).not.toContain('undefined')
  })
})

describe('generateFileHeader', () => {
  it('creates a flow type definition', () => {
    const rsp = generateFileHeader()

    expect(rsp).toMatchSnapshot()
    expect(rsp).not.toContain('undefined')
  })
})

describe('generator', () => {
  it('creates a hole flow type definition file output', () => {
    let schema = {
      services: [{
        name: 'GrpcService',
        methods: [{
          name: 'ReverseGeocode',
          input_type: 'ReverseGeocodeRequest',
          output_type: 'Location',
          client_streaming: false,
          server_streaming: false,
          options: {}
        }]
      }]
    }
    const rsp = generator(schema)

    expect(rsp).toMatchSnapshot()
    expect(rsp).not.toContain('undefined')
  })
})
