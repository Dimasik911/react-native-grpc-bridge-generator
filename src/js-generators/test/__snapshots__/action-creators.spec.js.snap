// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateActionCreator creates a repeated type definition 1`] = `
"export const getUser = (payload: GetUser) => ({
  meta: {
    delivery: 'grpc'
  },
  type: 'GET_USER_GRPC_REQUEST',
  payload
})"
`;

exports[`generateFileHeader creates a flow type definition 1`] = `"// @flow"`;

exports[`generator creates a hole flow type definition file output 1`] = `
"// @flow

import type { ReverseGeocodeRequest } from './flow-types'

export const reverseGeocode = (payload: ReverseGeocodeRequest) => ({
  meta: {
    delivery: 'grpc'
  },
  type: 'REVERSE_GEOCODE_GRPC_REQUEST',
  payload
})"
`;
